cmd/kubeadm/app/cmd/upgrade/diff.go

```diff
@@ -75,13 +81,22 @@ func NewCmdDiff(out io.Writer) *cobra.Command {
 }
 
 func runDiff(flags *diffFlags, args []string) error {
-
-	// If the version is specified in config file, pick up that value.
-	cfg, err := configutil.LoadInitConfigurationFromFile(flags.cfgPath)
+	var err error
+	var cfg *kubeadmapi.InitConfiguration
+	if flags.cfgPath != "" {
+		cfg, err = configutil.LoadInitConfigurationFromFile(flags.cfgPath)
+	} else {
```

Comment:
can you add a test case for this ?

Reply:
Thanks @yastij
This does lack test cases. But `FetchInitConfigurationFromCluster` requires a running kubernetes cluster. How should we write this kind of test cases? Any examples?

Reply:
If we want to unit test this, we need to mock `FetchInitConfigurationFromCluster`

Reply:
yeah, let's not do the mock/unit test of the function in question for now.

Reply:
I can settle with that, modulo a TODO or something to make us remember to do this

Reply:
@yastij Where should we put the `TODO:`? in `diff_test.go` ?

Reply:
`diff_test.go` seems reasonable to me

Reply:
@yastij Done